test/lifecycle_test.dart

```diff
@@ -429,6 +429,87 @@ main() {
         }, throwsA(anything));
       });
     });
+
+    group('Function Component', () {
+      Element mountNode;
+
+      setUp(() {
+        mountNode = DivElement();
+      });
+
+      group('', () {
+        ReactDartFunctionComponentFactoryProxy PropsTest;
+
+        setUpAll(() {
+          PropsTest = react.registerFunctionComponent((Map props) {
+            return [props['testProp'], props['children']];
+          });
+        });
+
+        tearDownAll(() {
+          PropsTest = null;
+        });
+
+        test('renders correctly', () {
+          var testProps = {'testProp': 'test'};
+          react_dom.render(PropsTest(testProps, ['Child']), mountNode);
+          expect(mountNode.innerHtml, 'testChild');
+        });
+
+        test('updates on rerender with new props', () {
+          var testProps = {'testProp': 'test'};
+          var updatedProps = {'testProp': 'test2'};
+          react_dom.render(PropsTest(testProps, ['Child']), mountNode);
+          expect(mountNode.innerHtml, 'testChild');
+          react_dom.render(PropsTest(updatedProps, ['Child']), mountNode);
+          expect(mountNode.innerHtml, 'test2Child');
+        });
+      });
+
+      group('recieves a JsBackedMap from the props argument', () {
+        var propTypeCheck;
+        ReactDartFunctionComponentFactoryProxy PropsArgTypeTest;
+
+        setUp(() {
+          PropsArgTypeTest = react.registerFunctionComponent((Map props) {
+            propTypeCheck = props;
+            return null;
+          });
+        });
+
+        tearDownAll(() {
```

Comment:
#nit this teardown isn't really necessary since `PropsArgTypeTest` is always overwritten in setUp